lib/util/flat-map.ts

```diff
@@ -1,11 +1,9 @@
-module.exports = flatMap
-
 // Note: `a` could be given here, which is then passed to the map function.
 // It functions as a rest/spread, but smaller.
-function flatMap(array, map, a) {
+export default function flatMap<T, A>(array: T[][], map: (items: T[], a: A) => T[], a?: A): T[] {
```

Comment:
what is `a`? and how is it used?

Reply:
It was needed for (actual) buffers, as an encoding can be passed. In theory, it is anything that is given to `map` as the second param, after the value. In practise, it’s a Node buffer character encoding (such as `'utf8'`)